app/controllers/companies_controller.rb

```diff
@@ -0,0 +1,74 @@
+class CompaniesController < ApplicationController
+  before_action :set_company, only: [:show, :edit, :update, :destroy]
+
+  # GET /companies
+  # GET /companies.json
+  def index
+    @companies = Company.all
+  end
+
+  # GET /companies/1
+  # GET /companies/1.json
+  def show
+  end
+
+  # GET /companies/new
+  def new
+    @company = Company.new
+  end
+
+  # GET /companies/1/edit
+  def edit
+  end
+
+  # POST /companies
+  # POST /companies.json
+  def create
+    @company = Company.new(company_params)
+
+    respond_to do |format|
+      if @company.save
+        format.html { redirect_to @company, notice: 'Company was successfully created.' }
```

Comment:
Line is too long. [89/80]<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.